src/include/daos/rpc.h

```diff
@@ -93,6 +93,22 @@ enum daos_rpc_type {
 	DAOS_REQ_TGT,
 };
 
+struct daos_req_comm_in {
+	uint32_t	req_in_uid;
+	uint32_t	req_in_gid;
+	uint32_t	req_in_projid;
+	uint64_t	req_in_enqueue_id;
+	uint64_t	req_in_paddings[4];
+	d_string_t	req_in_addr;
+	d_string_t	req_in_jobid;
+};
+
+struct daos_req_comm_out {
+	uint64_t	req_out_enqueue_id;
```

Comment:
is it a per server XS unique ID to identify retried OP? better with a few comments about what it is

I have not fully understood some details, a possible case for example a fetch request/rpc sent to rank 1's tag 1, and server cannot serve it and reply -DER_OVERLOAD_RETRY with an enqueue_id. But when client retry, that server dead and will do degraded fetch so the retried fetchh request send to rank 2's tag 2 with that enqueue_id got from rank 1's tag 1, will that cause problem? 
similar problem for UPDATE if the retired UPDATE goes to different rank/tag (due to pool map change).
One possible problem maybe that the enque_id got from rank 1 tag 1 is very big, when retry to rank 2 tag 2 it may cause the retried request cannot be served in time?

Reply:
In the client codes, enqueue id is per shard, not per object. probably fine? or i mis-write something